### ISSUE TITLE
Add thanks for contributor which is not for code

### ISSUE THREAD

**Issue by tkoyama010** [pyvista/pyvista#6333]
### Describe what maintenance you would like added.

In recent years, contributions to open source projects, including PyVista, have become more diverse. This includes things like creating and promoting talks and tutorials, and event planning, which don't show up in the code.

GeoVista has introduced the [allcontributors-bot](https://github.com/all-contributors/all-contributors), at the suggestion of @bjlittle. See, for example, the conversation at https://github.com/bjlittle/geovista/pull/948.

We already thank code contributors in [`AUTHORS.rst`](https://github.com/pyvista/pyvista/blob/main/AUTHORS.rst), but we suggest that we create a [`CONTRIBUTORS.md`](https://github.com/bjlittle/geovista/blob/main/CONTRIBUTORS.md) file in the same way as GeoVista and record non-code contributions there.

### Links to source code.

https://github.com/all-contributors/all-contributors

### Pseudocode or Screenshots

None

**Comment by adeak** [comment]
@pyvista/developers FYI, see also linked PR :)

**Comment by tkoyama010** [comment]
@adeak Thank you for your interest. This was inspired by the fact that other projects had added a document to thank contributors other than those who contributed code.
We already express our gratitude to contributors to the code as authors in `AUTHORS.rst`. However, contributing to OSS is not just that. For example,

1. @adeak really have actively reviewed code. 
1. @tkoyama010 is doing translating document in local language.
1. @bjlittle is making awesome external package.
1. I got some useful ideas about documentation from the Matplotlib project at SciPy2024.

These are very significant contributions, even though it doesn't appear in the commit history. We all know that it has helped our project to move forward, and we want people who will be joining our project in the future to know about it. Also, commenting on little ideas on GitHub and at conferences has brought a wealth of ideas to our project.
Recently, several projects have adopted [allcontributors](https://allcontributors.org/), but perhaps the following page from GeoVista is the easiest to imagine as a completed project.
https://geovista.readthedocs.io/en/latest/reference/about.html#contributors

Please see below for instructions on how to use it after it has been set up.

https://allcontributors.org/docs/en/bot/usage

The following are minor points of concern.
1. Is the mention of commands too annoying for the other person?
2. Does the other person want to be added?

For 1., it would be best to try to do as much of the additional work as possible in one go so that you only have to mention it once. For 2., it seems that people who dislike this have not been observed in other projects. However, if there is such a person, you should guarantee that you will revert.

**Comment by tkoyama010** [comment]
> To be clear it's a great idea to attribute all contributors, I just want to discuss the exact mechanics here. E.g. AUTHORS.md + CONTRIBUTORS.md feels redundant to me. I mean details such as this

The above is a comment I received in PR, but this is also one of my concerns. My opinion is that AUTHOR.rst is an important record, so I would like to continue using it alongside CONTRIBUTORS.md for a while. I think it would be best to delete it once it has been determined that it has completely fulfilled its role.

**Comment by adeak** [comment]
Thanks a lot for the details, @tkoyama010.

On first glance I find the generated document jarring, unfortunately. It looks even worse on mobile:
![Screenshot_20240713_181518_Opera](https://github.com/user-attachments/assets/b88b3888-1f7d-4b3e-9032-dfb6ff42dc9d)
But even if we fix the spacing issue I find it very hard to parse all these emoji. It would take me so much effort to figure out anything but the names that I wouldn't even start. And I'm sure that's also highly unaccessible: readers with cognitive issues or poor eyesight (especially: screen reader users) would be blocked from that page.

This is just a first impression for now, I'll keep prodding this. But I'd find much more value in a text-based list format. Either a list of people by contributor category, or a list of contributor category by people. And then we could merge the two markdown files. But I suspect the bot is not this well configurable...

**Comment by tkoyama010** [comment]
I have succeeded in changing my mindset. I decided to add a more concise text list in #6555. Thanks for the advice :+1:
